src/AspNetCoreModuleV2/RequestHandlerLib/PipeOutputManager.cpp

```diff
@@ -194,13 +188,19 @@ PipeOutputManager::ReadStdErrHandleInternal(
         }
         else if (GetLastError() == ERROR_BROKEN_PIPE)
         {
-            break;
+            return;
         }
     }
-}
 
-VOID
-PipeOutputManager::NotifyStartupComplete()
-{
-    StopOutputRedirection();
+    char tempBuffer[MAX_PIPE_READ_SIZE];
```

Comment:
I would use a new constant like `BUFFER_SIZE` rather than re-using `MAX_PIPE_READ_SIZE`, since these two sizes are unrelated and may not be the same going forward.  Specifically, I think `BUFFER_SIZE` will always be `4096`, but we may increase or decrease `MAX_PIPE_READ_SIZE` in the future.